README.adoc

```diff
@@ -4690,6 +4690,31 @@ ary[..42]
 ary[0..42]
 ----
 
+=== Collection querying [[collection-querying]]
+
+When possible, use https://docs.ruby-lang.org/en/master/Enumerable.html#module-Enumerable-label-Methods+for+Querying[predicate methods from `Enumerable`] rather than expressions with `#count`.
```

Comment:
You might also want to add a bit of rationale here (e.g. readability and better performance in some cases)

Reply:
Added a paragraph that explains readability and performance benefits